### PR TITLE
Adding build check for invalid BlazorWebAssemblyLazyLoad property

### PR DESCRIPTION
**Pull request by uabarahona**
#Summary

This add build check for invalid BlazorWebAssemblyLazyLoad property:

- [x] For Build
- [x] For Publish
- [x] Add tests

Note: The check are added just before `BlazorWebAssemblyLazyLoad` if your think the error should be checked as soon as possible I can spend a bit more time figuring out where could be the best place 

Addresses #24880

### CODE REVIEW

File: src/Components/WebAssembly/Sdk/integrationtests/WasmBuildLazyLoadTest.cs [hunk 1]
```diff
@@ -165,6 +165,62 @@ public async Task Publish_LazyLoadExplicitAssembly_Release_Works()
...
+            // Act
+            var result = await MSBuildProcessManager.DotnetMSBuild(project, "Publish");
+
+            // Assert
+            Assert.BuildError(result, "BLAZORSDK1001");
+            Assert.BuildFailed(result);
+        }
+
+        [Fact]
+        public async Task Build_Without_LazyLoadExplicitAssembly_Works()
```
Comment (captainsafia) [inline src/Components/WebAssembly/Sdk/integrationtests/WasmBuildLazyLoadTest.cs:211]: I don't think this test is necessary. It's covered by our publish tests and we have coverage on a standard lazy-load scenario as well.

File: src/Components/WebAssembly/Sdk/src/targets/Microsoft.NET.Sdk.BlazorWebAssembly.Current.targets [hunk 2]
```diff
@@ -283,6 +283,11 @@ Copyright (c) .NET Foundation. All rights reserved.
       <_BlazorBuildBootJsonPath>$(IntermediateOutputPath)blazor.boot.json</_BlazorBuildBootJsonPath>
     </PropertyGroup>
 
+    <Error
+      Text="Unable to find $(OutDir)$(_BlazorOutputPath)%(BlazorWebAssemblyLazyLoad.Identity) to be lazy loaded later (are you missing an assembly reference?)"
```
Comment (captainsafia) [inline src/Components/WebAssembly/Sdk/src/targets/Microsoft.NET.Sdk.BlazorWebAssembly.Current.targets:287]:
```suggestion
      Text="Unable to find $(OutDir)$(_BlazorOutputPath)%(BlazorWebAssemblyLazyLoad.Identity) to be lazy loaded later. Confirm that project or package references are included and the reference is used in the project."
```

This seems like a more helpful error message.

**Review by captainsafia — COMMENTED** [review]
Thanks! Left two small comments inline.

File: src/Components/WebAssembly/Sdk/integrationtests/WasmBuildLazyLoadTest.cs (hunk 1 above)
Comment (barahonajm) [reply src/Components/WebAssembly/Sdk/integrationtests/WasmBuildLazyLoadTest.cs:211]: Go it, I will change it

File: src/Components/WebAssembly/Sdk/src/targets/Microsoft.NET.Sdk.BlazorWebAssembly.Current.targets (hunk 2 above)
Comment (barahonajm) [reply src/Components/WebAssembly/Sdk/src/targets/Microsoft.NET.Sdk.BlazorWebAssembly.Current.targets:287]: Agree, will change it

**Review by captainsafia — APPROVED** [review]
Looks good. Thanks!

**Pull request merged**